src/diagnostic/detail/types/particle.h

```diff
@@ -62,6 +67,26 @@ class ParticlesDiagnosticWriter : public H5TypeWriter<HighFiveDiagnostic>
 };
 
 
+template<typename HighFiveDiagnostic>
+void ParticlesDiagnosticWriter<HighFiveDiagnostic>::createFiles(DiagnosticProperties& diagnostic)
+{
+    auto& hi5 = this->hi5_;
+
+    auto checkCreate = [&](auto& tree, auto var) {
```

Comment:
should that lambda somehow be factored out (same as in fluid?)... no pressure... maybe more readable as is than isolated somewhere else?

Reply:
not sure I get this

Reply:
sorry :D 
I meant it looks like this lambda is the same as for the fluid create function, hence maybe could be a function used in both cases rather than duplicating.

Reply:
ok can do